infrastructure/base-container-img

```diff
@@ -0,0 +1,39 @@
+FROM ubuntu:24.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV DEVCONTAINER_ENV=1
+ENV BUNDLE_PATH=/usr/local/bundle
+ENV BUNDLE_APP_CONFIG=/usr/local/bundle
+
+# Set working directory
+WORKDIR /workspace
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    git \
+    gh \
+    less \
+    python3 \
+    python3.12-venv \
+    python3-pip \
+    build-essential \
+    ruby \
+    ruby-dev \
+    bundler \
+    nodejs \
+    npm \
+    ditaa && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy and install Ruby dependencies
+COPY Gemfile Gemfile.lock ./
```

Comment:
We've been installing the Ruby/Node/Python packages on the host (under `${UDB_ROOT}/.home/.gems`, `${UDB_ROOT}/node_modules`, and `${UDB_ROOT}/.home/.venv`, respectively). We then make sure that the UDB_ROOT is bound in the container. The thought was that it would (a) keep the container smaller, and (b) make it easier to add packages as we go (most additions won't need a container rebuild). You can see how all this gets set up in `bin/setup` (which is implicitly called any time you execute a program under `bin/` or the top-level `do` script).

Is there a reason that we should be including them in the container instead?